main.go

```diff
@@ -265,7 +265,9 @@ func main() {
 		BurstSize: 100,
 	})
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	restConfig := ctrl.GetConfigOrDie()
+	restConfig.UserAgent = "capz-controller"
```

Comment:
```suggestion
	restConfig.UserAgent = "cluster-api-provider-azure-controller"
```
Suggesting this because capa has similar naming => https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2285

Reply:
@shysank hi! Thanks for the suggestion, fixed it!